src/Tools/dotnet-monitor/CollectionRules/Actions/CollectTraceAction.cs

```diff
@@ -83,7 +85,18 @@ public CollectTraceAction(IServiceProvider serviceProvider, IEndpointInfo endpoi
                 KeyValueLogScope scope = Utils.CreateArtifactScope(Utils.ArtifactType_Trace, EndpointInfo);
 
                 EgressOperation egressOperation = new EgressOperation(
-                    (outputStream, token) => TraceUtilities.CaptureTraceAsync(startCompletionSource, EndpointInfo, configuration, duration, outputStream, token),
+                    async (outputStream, token) =>
+                    {
+                        IDisposable operationRegistration = _operationTrackerService.Register(EndpointInfo);
```

Comment:
nit: Because this is unconditionally created, you could use a `using IDisposable registration = ` and not use a try...finally. I guess when compiled, it does the same thing.